extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -226,13 +226,11 @@ export class AmpStoryPage extends AMP.BaseElement {
     /** @private @const {!../../../src/service/resources-impl.Resources} */
     this.resources_ = Services.resourcesForDoc(getAmpdoc(this.win.document));
 
-    /** @private @const {!Promise} */
-    this.mediaLayoutPromise_ = this.waitForMediaLayout_();
+    /** @private {?Promise} */
+    this.mediaLayoutPromise_ = null;
 
-    /** @private @const {!Promise} */
-    this.pageLoadPromise_ = this.mediaLayoutPromise_.then(() => {
-      this.markPageAsLoaded_();
-    });
+    /** @private {?Promise} */
+    this.pageLoadPromise_ = null;
```

Comment:
This is going to cause race conditions.  This is exposed through the public method `whenLoaded` and we don't (or perhaps shouldn't) have the guarantee that `whenLoaded` will be called after `layoutCallback`.  As such, `whenLoaded` will return `null` initially, which violates the contact of returning a Promise that resolves when the page has loaded.

You can instead try following the deferred pattern, like `this.mediaPoolPromise_` in this class.